collections/sort_by.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Returns all elements in the given collection, sorted by their result using the given selector
+ *
+ * Example:
+ *
+ * ```typescript
+ * const people = [
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ *     { name: 'John', age: 23 },
+ * ]
+ * const sortedByAge = sortBy(people, it => it.age)
+ *
+ * console.assert(sortedByAge === [
+ *     { name: 'John', age: 23 },
+ *     { name: 'Anna', age: 34 },
+ *     { name: 'Kim', age: 42 },
+ * ])
+ * ```
+ */
+export function sortBy<T>(
+  array: Array<T>,
+  selector:
+      | ((el: T) => number)
+      | ((el: T) => string)
+      | ((el: T) => bigint)
+      | ((el: T) => boolean)
+      | ((el: T) => Date),
```

Comment:
I like this idea of restricting the return value :+1: This should reduce the confusion in the user's code base.

But I'm not sure if we should include boolean as returned value. Do you think there is a use case for that? If the purpose of sorting by boolean is to separate items in 2 groups, how about encouraging the use of `partition`?

Reply:
I have no strong opinion on this - I just included the builtins / primitives that get sorted "correctly" by the spec. I agree that `boolean` might be less useful. If no one objects, I will remove it.

Reply:
I removed it